mlir/lib/Dialect/Rock/Transforms/GemmToGridwise.cpp

```diff
@@ -166,17 +187,55 @@ GemmRewritePattern::matchAndRewrite(GemmOp op, GemmOpAdaptor adaptor,
   if (!gridSize)
     return op.emitOpError("grid size must be set at lowering");
 
+  auto aElementType = a.getType().cast<MemRefType>().getElementType();
+  auto bElementType = b.getType().cast<MemRefType>().getElementType();
+  auto cElementType = c.getType().cast<MemRefType>().getElementType();
+
+  auto accumulatorElementType = obtainAccumulatorElementType(
+      rw, aElementType, bElementType, cElementType);
+  auto accumulatorShape = c.getType().cast<MemRefType>().getShape();
+  auto accumulatorType =
+      MemRefType::get(accumulatorShape, accumulatorElementType);
+  auto accumulator = rw.create<memref::AllocOp>(loc, accumulatorType);
+
   if (isAccel) {
     rw.create<GridwiseGemmAccelOp>(
-        loc, a, b, c, op.getArchAttr(), numCUAttr, op.getFeaturesAttr(),
-        op.getStoreMethodAttr(), blockSize, gridSize,
+        loc, a, b, accumulator, op.getArchAttr(), numCUAttr,
+        op.getFeaturesAttr(), op.getStoreMethodAttr(), blockSize, gridSize,
         params.cast<RockAccelTuningParamAttrInterface>());
-    rw.eraseOp(op);
   } else {
-    rw.create<GridwiseGemmOp>(loc, a, b, c, op.getFeaturesAttr(), numCUAttr,
-                              gridSize, params.cast<GeneralGemmParamsAttr>());
-    rw.eraseOp(op);
+    rw.create<GridwiseGemmOp>(loc, a, b, accumulator, op.getFeaturesAttr(),
+                              numCUAttr, gridSize,
+                              params.cast<GeneralGemmParamsAttr>());
   }
+
+  AffineExpr g = rw.getAffineDimExpr(0), m = rw.getAffineDimExpr(1),
+             n = rw.getAffineDimExpr(2);
+
+  AffineMap cMap = AffineMap::get(3, 0, {g, m, n}, rw.getContext());
```

Comment:
If you look into `AlignTiling`, there's a function that gets called there for getting a multi-dimensional identity map.

Reply:
Cool. Thank for the advice!
Done